main.js

```diff
@@ -0,0 +1,260 @@
+  const selectedPost =
+    event.target.parentElement.parentElement.parentElement.parentElement;
+  const commentSectionId = selectedPost
+    .querySelector(".card")
+    .parentElement.getAttribute("id");
+  const postNumber = commentSectionId.match(/\d+$/)[0];
+
+  // Get comment data
+  const commentContent = document.querySelector(
+    `#post-comment-${postNumber}`
+  ).value;
+  const userName = document.querySelector(`#your-name-${postNumber}`).value;
+
+  if (commentContent === "" || userName === "") {
+    throw error;
+  }
+
+  const comment = {
+    text: commentContent,
+    name: userName,
+  };
+
+  // Push comment to post object
+  postList[postNumber].comments.push(comment);
+
+  // Update UI
+  const commentDiv = document.createElement("div");
+  commentDiv.setAttribute("id", `comment-number-${commentNum}`);
+  commentDiv.setAttribute("class", "single-comment");
+  const commentText = document.createTextNode(comment.text);
+
+  commentDiv.appendChild(commentText);
+
+  const lineBreak = document.createElement("br");
+
+  commentDiv.appendChild(lineBreak);
+
+  const postedByText = document.createTextNode("Posted by: ");
+
+  commentDiv.appendChild(postedByText);
+
+  const boldText = document.createElement("b");
+  const name = document.createTextNode(comment.name);
+
+  boldText.appendChild(name);
+
+  commentDiv.appendChild(boldText);
+
+  const hrDivider = document.createElement("hr");
+
+  document
+    .querySelector(`.post-comment-section-${postNumber}`)
+    .append(commentDiv);
+  document
+    .querySelector(`.post-comment-section-${postNumber}`)
+    .append(hrDivider);
+
+  // Delete comment button
+  document.querySelector(
+    `#comment-number-${commentNum}`
+  ).innerHTML += `<span class="icon-button" data-bs-toggle="modal" data-bs-target="#delete-comment-modal"><i class="fa-solid fa-trash-can remove-comment"></i></span>`;
+
+  // Clear message box
+  document.querySelector(`#your-name-${postNumber}`).value = "";
+  document.querySelector(`#post-comment-${postNumber}`).value = "";
+
+  // Increment number of comments
+  commentNum++;
+};
+
+const removePost = (event) => {
+  // Add delete button event listener
+  const deletePostButton = document.querySelector(".delete-post-button");
+
+  deletePostButton.addEventListener("click", deletePost);
+
+  // Delete selected post
+  function deletePost() {
+    document.querySelector('[data-bs-dismiss="modal"]').click();
+    event.target.closest(".single-post").nextSibling.remove();
+    event.target.closest(".single-post").remove();
+    removeHandler();
+  }
+
+  function removeHandler() {
+    deletePostButton.removeEventListener("click", deletePost);
+  }
+};
+
+const removeComment = (event) => {
+  // Add delete button event listener
+  const deleteCommentButton = document.querySelector(".delete-comment-button");
+
+  deleteCommentButton.addEventListener("click", deleteComment);
+
+  // Delete selected comment
+  function deleteComment() {
```

Comment:
This is really repetitive and could be moved to a separate function outside of this that could be used by removePost as well - would just need to pass in all the details that need to make it less specific. 

Also not really best practice to write functions inside of functions like this - if you want to call a function inside a function, its better to pass it in as a parameter and then call in the function.

Would also recommend not using function declarations and instead use function expressions.